### PR TITLE
Ruby: add support for one-line pattern matches

### PR DESCRIPTION
**Pull request by aibaars**

### CODE REVIEW

File: ruby/ql/lib/codeql/ruby/ast/internal/AST.qll [hunk 1]
```diff
@@ -130,6 +134,7 @@ private module Cached {
     TDoBlock(Ruby::DoBlock g) { not g.getParent() instanceof Ruby::Lambda } or
     TElementReference(Ruby::ElementReference g) or
     TElse(Ruby::Else g) or
+    TElseSynth(Ast::AstNode parent, int i) { mkSynthChild(ElseKind(), parent, i) } or
```
Comment (hvitved) [inline ruby/ql/lib/codeql/ruby/ast/internal/AST.qll:137]: Should we rename the existing `TElse` to `TElseReal` and define `class TElse = TElseReal or TElseSynth;`?

File: ruby/ql/lib/codeql/ruby/ast/Control.qll [hunk 2]
```diff
@@ -545,6 +541,88 @@ class InClause extends AstNode, TInClause {
...
+  private Ruby::MatchPattern g;
+
+  MatchPattern() { this = TMatchPattern(g) }
+
+  final override string getAPrimaryQlClass() { result = "MatchPattern" }
+
+  /**
+   * Gets the expression being compared, if any. For example, `foo` in the following example.
+   * ```rb
+   * case foo => Point{ x:, y: }
```
Comment (hvitved) [inline ruby/ql/lib/codeql/ruby/ast/Control.qll:560]: Copy-paste QL doc?

File: ruby/ql/lib/codeql/ruby/ast/Control.qll [hunk 3]
```diff
@@ -545,6 +541,88 @@ class InClause extends AstNode, TInClause {
...
+  private Ruby::TestPattern g;
+
+  TestPattern() { this = TTestPattern(g) }
+
+  final override string getAPrimaryQlClass() { result = "TestPattern" }
+
+  /**
+   * Gets the expression being compared, if any. For example, `foo` in the following example.
+   * ```rb
+   * case foo in Point{ x:, y: }
```
Comment (hvitved) [inline ruby/ql/lib/codeql/ruby/ast/Control.qll:601]: Same

File: ruby/ql/lib/codeql/ruby/ast/Control.qll [hunk 4]
```diff
@@ -545,6 +541,88 @@ class InClause extends AstNode, TInClause {
...
+   * ```
+   */
+  final Expr getValue() { toGenerated(result) = g.getValue() }
+
+  /**
+   * Gets the pattern in this `=>` expression. In the
```
Comment (hvitved) [inline ruby/ql/lib/codeql/ruby/ast/Control.qll:607]: Copy-paste error

**Review by hvitved — CHANGES_REQUESTED** [review]
Looks great! Only some trivial comments. Also, remember a DCA run.

File: ruby/ql/lib/codeql/ruby/ast/internal/AST.qll (hunk 1 above)
Comment (aibaars) [reply ruby/ql/lib/codeql/ruby/ast/internal/AST.qll:137]: We could, but not sure if adding `class TElse = TElseReal or TElseSynth` is useful.

Comment (hvitved) [reply ruby/ql/lib/codeql/ruby/ast/internal/AST.qll:137]: I think it is nice for consistency.

File: ruby/ql/lib/codeql/ruby/ast/Control.qll [hunk 5]
```diff
@@ -557,7 +557,7 @@ class MatchPattern extends Expr, TMatchPattern {
   /**
    * Gets the expression being compared, if any. For example, `foo` in the following example.
```
Comment (hvitved) [inline ruby/ql/lib/codeql/ruby/ast/Control.qll:558]: Remove `if any`?

File: ruby/ql/lib/codeql/ruby/ast/Control.qll [hunk 6]
```diff
@@ -598,13 +598,13 @@ class TestPattern extends Expr, TTestPattern {
...
```
Comment (hvitved) [inline ruby/ql/lib/codeql/ruby/ast/Control.qll:599]: Remove `if any`?

**Review by hvitved — APPROVED** [review]

**Pull request merged**